numba/tests/test_literal_dispatch.py

```diff
@@ -0,0 +1,254 @@
+from __future__ import print_function
+
+import numpy as np
+
+import numba
+import numba.unittest_support as unittest
+from numba.tests.support import TestCase
+from numba import njit, types, errors
+from numba.extending import overload, SentryLiteralArgs
+from numba.special import literally
+
+
+class TestLiteralDispatcher(TestCase):
+    def test_literal_basic(self):
+        @njit
+        def foo(x):
+            return literally(x)
+
+        self.assertEqual(foo(123), 123)
+        self.assertEqual(foo(321), 321)
+
+        sig1, sig2 = foo.signatures
+        self.assertEqual(sig1[0].literal_value, 123)
+        self.assertEqual(sig2[0].literal_value, 321)
+
```

Comment:
Most `Literal[T]` are just subclasses of `T` and uses the unboxer from `T`. However, `Literal[str]` is different because of py2 vs py3.  I'll add an explicit unboxer.

Reply:
thanks